### ISSUE TITLE
Reparenting deletes nodes in inherited scenes

### ISSUE THREAD

**Issue by fibr0bl4st** [godotengine/godot#38186]
**Godot version:**
3.1.1stable
**OS/device including version:**
Win 10 1809
**Issue description:**
When inheriting a scene moving or reparenting (scene tree) children in the (object-oriented) parent,
related overwritten or added nodes in the (object-oriented) child are being deleted without a warning.
Since you see no change in the (object-oriented) children without saving the parent scene, when you notice this behavior you are unable to use the undo function and have already lost your work.
**Steps to reproduce:**
1) Create a simple scene with any two child nodes and save: (tested with Node2D and Sprite)
root
|- node1
|- node2
2) Create a New Inherited Scene and make any changes to node2 and save.
Add Children change properties anything goes.
3) Go to the original Scene (object-oriented parent), reparent node2 to as a child of node1 and save:
root
|- node1
&nbsp;&nbsp; |- node2
4) Go to the inherited scene. Your previous changes are lost.
The debug-console should read:
ERROR: Node not found: nodename

**Minimal Project:**
In this Project you have already completed step 2)
[Test.zip](https://github.com/godotengine/godot/files/4531042/Test.zip)

**Possible solutions:**
Most obvious is to move the node in the inherited nodes as well.
Another good behavior would be to either dublicate the node in question, so the user has a chance to transfer the changes by hand, or to show a warning, when reparenting something in an (object-oriented) parent scene.

**Personal comment:**
Just lost a days work to that. Why did I not use my versioning tool :pensive:

**Comment by fibr0bl4st** [comment]
After more research I fear this issue might be a duplicate to
https://github.com/godotengine/godot/issues/16650
Sorry for not noticing that issue earlier.

For other people having the same problem:
I was able to rescue part of my lost work.
In another scene I had instanced the described child scenes and in there, one was not yet updated.
I could "make local" them and with that copy most of their parameters back into the original scene.
